### DISCUSSION TITLE
Merging json objects

### DISCUSSION THREAD

**Discussion by Batlan** [nlohmann/json#2847] [Q&A] [answered]
Hi,
I want to merge two json objects and need to ensure that the existing object is updated if a corresponding key exists in the second object.
Looking through the documentation, I found exactly what I was looking for, namely the `update(j)` method.

Unfortunately when using it, I realised that it doesn't fulfill my expectations. This is due to the fact that the `update(j)` method replaces the entire key if it is found in the update structure, instead of iterating over it in case this key contains a json subtree.

Example:
```
 json o1 = R"( {"one": "an element", "two": {"name": "an object", "type": "object", "aka": "glipglop"}, "three": 3})"_json;
 json o2 = R"( {"one": "an elephant", "two": {"name": "Fred", "type": "fred"}, "four": 4})"_json;
 o1.update(o2);
std::cout << o1 << '\n';
```
results in 
```
{“one”: “an elephant”, “two”: {“name”: “Fred”, “type”: “fred”}, “three”: 3, “four”: 4}
```
instead of the expected
```
{“one”: “an elephant”, “two”: {“name”: “Fred”, “type”: “fred”, "aka": "glipglop"}, “three”: 3, “four”: 4}
```

Is the current result the expected behaviour of the `update(j)` function? If yes, what's the best way to achieve my expected result?

**Comment by nlohmann** [comment, marked as answer]
Right now, `update` is not implemented recursively: https://github.com/nlohmann/json/blob/develop/include/nlohmann/json.hpp#L5906-L5929

I think your expected behavior would be possible by recursively call `update` on all children.

**Reply by nlohmann** [reply]
Do you need further assistance with this issue?

**Reply by Batlan** [reply]
Thanks for fast reply at the time.
Unfortunately I thought I could quick fix a bug with the update function, now I will have to do it implement it myself using your suggestion and test it more thoroughly  :-)
